packages/accounts-base/accounts_server.js

```diff
@@ -1115,6 +1115,41 @@ Ap._expireTokens = function (oldestValidDate, userId) {
   // expired tokens.
 };
 
+// Deletes expired password reset tokens from the database.
+//
+// Exported for tests. Also, the arguments are only used by
+// tests. oldestValidDate is simulate expiring tokens without waiting
+// for them to actually expire. userId is used by tests to only expire
+// tokens for the test user.
+Ap._expirePasswordResetTokens = function (oldestValidDate, userId) {
+  var tokenLiftetimeMs = this. _getPasswordResetTokenLifetimeMs();
+
+  // when calling from a test with extra arguments, you must specify both!
+  if ((oldestValidDate && !userId) || (!oldestValidDate && userId)) {
+    throw new Error("Bad test. Must specify both oldestValidDate and userId.");
+  }
+
+  oldestValidDate = oldestValidDate ||
+    (new Date(new Date() - tokenLifetimeMs));
+  var userFilter = userId ? {_id: userId} : {};
+
+  this.users.update(_.extend(userFilter, {
+    $or: [
+      { "services.password.reset.when": { $lt: oldestValidDate } },
+      { "services.password.reset.when": { $lt: +oldestValidDate } }
+    ]
+  }), {
+    $pull: {
+      "services.password.reset": {
+        $or: [
+          { when: { $lt: oldestValidDate } },
+          { when: { $lt: +oldestValidDate } }
+        ]
+      }
+    }
+  }, { multi: true });
+}
+
```

Comment:
I think there's an issue here if 'services.password.reset' isn't an array. Not sure why because it looks like it should only update if 'services.password.reset.when' exists. I'm running Meteor 1.4.0.1 but have updated accounts-base to 1.2.10 (maybe this is the issue?) Anyway I'm seeing a few errors in production:

```
MongoError: Cannot apply $pull to a non-array value ...[snip]... at AccountsServer.Ap._expirePasswordResetTokens
```